File: 2017/01/01.go
package main

import (
	"fmt"
	"io/ioutil"
	"strconv"
)

func main() {
	ints := InputBytesToInts("input")

	fmt.Println(partOne(ints))
	fmt.Println(partTwo(ints))
}

func InputBytesToInts(fileName string) []int {
	input, err := ioutil.ReadFile(fileName)
	if err != nil {
		panic(err)
	}
	var ints []int

	for _, s := range input {
		v, err := strconv.Atoi(string(s))
		if err != nil {
			panic(err)
		}
		ints = append(ints, v)
	}
	return ints
}

func partOne(ints []int) int {
	sum := 0

	for i := 0; i < len(ints); i++ {
		current := ints[i]
		next := ints[(i+1)%len(ints)]

		if current == next {
			sum += current
		}
	}

	return sum
}

func partTwo(ints []int) int {
	// List has an even number of elements
	steps := len(ints) / 2

	sum := 0

	for i := 0; i < len(ints); i++ {
		current := ints[i]
		next := ints[(i+steps)%len(ints)]

		if current == next {
			sum += current
		}
	}

	return sum
}

File: 2020/01/01.py

def solve1(ints):
    for i in range(len(ints)):
        j = i + 1
        while j < len(ints):
            if ints[i] + ints[j] == 2020:
                return ints[i] * ints[j]
            j += 1

def solve2(ints):
    for i in range(len(ints)):
        j = i + 1
        while j < len(ints):
            k = j + 1
            while k < len(ints):
                if ints[i] + ints[j] + ints[k] == 2020:
                    return ints[i] * ints[j] * ints[k]
                k += 1
            j += 1

def main():
    with open('input', 'r') as f:
        ints = [int(x) for x in f.readlines()]

    print(solve1(ints))
    print(solve2(ints))

main()